### ISSUE TITLE
Using frozen_string_literals still triggers "Freeze mutable objects assigned to constants."

### ISSUE THREAD

**Issue by qnet-herwin** [bbatsov/rubocop#4213]
Running rubocop of the following file

```ruby
# frozen_string_literal: true

STR = 'foo'
```

Generates a warning:

```
frozen.rb:3:7: C: Freeze mutable objects assigned to constants.
```

It would be pretty redundant to freeze them twice.

## RuboCop version

```
qnet@devel-o9:~$ ruby2.4 -S rubocop -V
0.48.0 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-linux)
```

**Comment by backus** [comment]
Do you have a `.ruby-version` file or a `TargetRubyVersion` in your config file? This usually means rubocop thinks you are using a ruby version which is 2.2 or lower and therefore doesn't support the frozen string literal magic comment

**Comment by qnet-herwin** [comment]
Well, that resolved the problem. Sorry for the case.

**Comment by pablox-cl** [comment]
I have both: 
```yaml
# .rubocop.yml
AllCops:
  TargetRubyVersion: 2.5.1
```
and

    2.5.1
on `.ruby-version`, and I'm still getting this issue. I'm doing something wrong?

```
❯ rubocop -V
0.59.2 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-linux)
```

**Comment by clive-hetzner** [comment]
I too am getting this on ruby 2.5.1, but only on Linux and not on OSX(darwin) The same code in our project, produces rubocop failures only on the linux dev machines and the CI (linux) - but not on mac osx machines.

**Comment by clive-hetzner** [comment]
```
bash-4.4# cat test.rb
# frozen_string_literal: true

A = 'hello world'
p A.frozen?
bash-4.4# rubocop test.rb
Inspecting 1 file
C

Offenses:

test.rb:3:5: C: Style/MutableConstant: Freeze mutable objects assigned to constants.
A = 'hello world'
    ^^^^^^^^^^^^^

1 file inspected, 1 offense detected
bash-4.4# ruby test.rb
true
bash-4.4# ruby --version
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux]
bash-4.4# rubocop --version
0.68.1
bash-4.4# 
```
